modulemd/modulemd-util.c

```diff
@@ -145,3 +145,100 @@ modulemd_variant_unref (void *ptr)
 {
   g_variant_unref ((GVariant *)ptr);
 }
+
+gboolean
+modulemd_validate_nevra (const gchar *nevra)
+{
+  g_autofree gchar *tmp = g_strdup (nevra);
+  gsize len = strlen (nevra);
+  gchar *i;
+  gchar *endptr;
+
+  /* Since the "name" portion of a NEVRA can have an infinite number of
+   * hyphens, we need to parse from the end backwards.
+   */
+  i = &tmp[len - 1];
+
+  /* Everything after the last '.' must be the architecture */
+  while (i >= tmp)
+    {
+      if (*i == '.')
+        {
+          break;
+        }
+      i--;
+    }
+
+  if (i < tmp)
+    {
+      /* We hit the start of the string without hitting '.' */
+      return FALSE;
+    }
+
+  /*
+   * TODO: Compare the architecture suffix with a list of known-valid ones
+   * This needs to come from an external source that's kept up to date or
+   * this will regularly break.
+   */
+
+  /* Process the "release" tag */
+  while (i > 0)
+    {
+      if (*i == '-')
+        {
+          break;
+        }
+      i--;
+    }
+
+  if (i <= 0)
+    {
+      /* We hit the start of the string without hitting '-' */
+      return FALSE;
+    }
+
+  /* No need to validate Release; it's fairly arbitrary */
+
+  /* Process the version */
+  while (i >= tmp)
+    {
+      if (*i == ':')
+        {
+          break;
+        }
+      i--;
+    }
+  if (i <= 0)
+    {
+      /* We hit the start of the string without hitting ':' */
+      return FALSE;
+    }
+
+  /* Process the epoch */
```

Comment:
epoch is delemited by `:`

Reply:
Yes, and? Did you see some mistake here?